### ISSUE TITLE
CSS: Инкапсулировать все стили в @layer

### ISSUE THREAD

**Issue by ufadev** [consta-design-system/uikit#3463]
**Описание фичи или улучшения**
Перевести все стили uikit в CSS layer

**Почему улучшение или фича необходима?**
При использовании @consta/uikit в проектах с микрофронтовой архитектурой (module federation) наблюдается проблема с перекрытием предопределенных стилей. Связано это с тем, что с микрофронтом приезжает часть бандлов консты, не используемых в хостовом приложении, ломая переопределения схожей специфичности.

В настоящий момент проблема решается только избыточным увеличением специфичности, и/или important!

**Реализация предложения**
Использовать [CSS layer](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer) для всех стилей библиотеки. Это позволит содержать стили компонентов и стили приложения (в том числе, переопределяющие библиотечные) на разных слоях с разным приоритетом.

**Comment by ya-lops** [comment]
Да, было бы очень полезно использовать и видеть нечто подобное:
`
@layer consta.combobox { ... }
`

**Comment by gizeasy** [comment]
протестировал `@layer`

Эксперементы и мысли:
- нужно иметь 1 слой для всех библиотек  `consta`, за тем чтобы можно было удобно указать его порядок в проекте
```
@layer reset consta overide
```
- Если хотим использовать `@layer` внутри компонентов то, для того чтобы слои библиотек не пересекались нужно добавить подслои для каждой билиотеки
```
@layer consta.libName
@layer consta.uikit
```
- нельзя указывать порядок слоев на глобальном уровне так можем афектить порядок проекта, можно указывать порядок только внутри слоя  `consta.libName`.
- также для инкапсуляции слоев компонетов стоит сделать отдельный слой для каждого компонента
```
@layer consta.uikit.Button
```
- возможно стоит добавить еще слои для components  и mixs так чтобы components всегда переопределяли mix
```
@layer consta.uikit.components.Button
@layer consta.uikit.mixs.Button
```
но это не значит что components из другой библиотеки будут переопределять mixs из uikit, можно попробовать по другому: выделить слои по назначению их вынести на верхний уровень `consta`
```
@layer consta.components.uikit.Button
@layer consta.mixs.uikit.Button
```
или пойти еще дальше и раздробить библиотеки по назначнию и уже библиотекам задать порядок переопределения, но я думаю что гденибуть не угалаем cвыбором.

и самая большая проблема (если мы на каждый компонент создаем слой) вместо того чтобы помогать переопределять мы только усложняем

```css
@layer consta.uikit.User{
 .Avatar.User-Avatar{
 /// тут хотим что-то переопределить, а не выходит так как слой 
 }
}

@layer consta.uikit.Avatar{
 .Avatar{

 }
}
```
В компоненте User мы хотим переопределить стили Avatar, но не выходит так как Avatar ниже(главнее),
но мы же можем все блоки которые использвются в User указать менее приорететными.
```
@layer consta.uikit {
  @layer Avatar, Text, MixSpace, User;
}
```

но тут вдруг в другом компонене нам потребовалось 
```
@layer consta.uikit {
  @layer Text, Avatar;
}
```

и получается что мы афектим порядок слоев

---

Вообщем по итогу - делить слои по компонентам смысла нет, у нас бывают случаи переопределения стилей но мы обходимся обычными селекторами в случае переопределения одного блока другим. Заводить еще термины как тип блока (components или mix) тоже не вижу смысла усложнять, пусть они все останутся равноправными блоками.

Выносить всю коснту в слой consta тут скорее да чем нет, но только в мажоре, так как стили у проектов могут поехать бональными стилями из "reset", так как слой будет иметь меньший приоритет.

сейчас на проеткте можно при сборке все файлов из `node_modules/@consta` оборачивать в @layer consta

**Comment by ya-lops** [comment]
возможно, да. избыточно

В качестве примера по теме: ...
в tailwind 4 очень хорошо реализованы слои внутри либы, но там и как таковых компонентов нет, точнее они в другом виде.
`@layer tailwind` и в нем вложенный порядок: `properties, theme, base, components, utilities` и тейлвинд никогда по умолчанию не меняет стили пользователя всякими `mt-4` потому что это утилита и имеет наименьший "вес". оч удобно
